apiserver/facades/client/resources/repository_test.go

```diff
@@ -149,6 +143,41 @@ func (s *CharmHubClientSuite) TestResolveResourcesUpload(c *gc.C) {
 	}})
 }
 
+func (s *CharmHubClientSuite) TestResourceInfo(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+	s.expectRefreshWithRevision(25, false)
+
+	curl := charm.MustParseURL("ch:amd64/focal/ubuntu-19")
+	rev := curl.Revision
+	channel := corecharm.MustParseChannel("stable")
+	origin := corecharm.Origin{
+		Source:   corecharm.CharmHub,
+		Type:     "charm",
+		Revision: &rev,
+		Channel:  &channel,
+		Platform: corecharm.Platform{
+			OS:           "ubuntu",
+			Series:       "focal",
+			Architecture: "amd64",
+		},
+	}
+
+	result, err := s.newClient().ResourceInfo(curl, origin, "wal-e", 25)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result, gc.DeepEquals, charmresource.Resource{
+		Meta:        charmresource.Meta{Name: "wal-e", Type: 1, Path: "wal-e.snap", Description: "WAL-E Snap Package"},
+		Origin:      charmresource.OriginStore,
+		Revision:    25,
+		Fingerprint: fp(),
+		Size:        0,
+	})
+}
+
+func fp() charmresource.Fingerprint {
+	fp, _ := charmresource.ParseFingerprint("38b060a751ac96384cd9327eb1b1e36a21fdb71114be07434c0cc7bf63f6e1da274edebfe76f65fbd51ad2f14898b95b")
```

Comment:
I think it's worth keeping the assert on the error. If, for whatever reason, this parse fails, it could be difficult to track down